src/app/workflow/workflow.component.ts

```diff
@@ -95,16 +111,56 @@ export class WorkflowComponent implements OnInit {
     });
   }
 
-  get numFiles(): number {
-    return this.study ? this.study.files.length : 0;
-  };
-
   ngOnInit(): void {
     window[`angularComponentReference`] = {
       component: this, zone: this.ngZone, loadAngularFunction: (str: string) => this.angularFunctionCalled(str),
     };
   }
 
+  updateWorkflow(wf: Workflow) {
+    // Do these things whenever the workflow changes due to an API call
+    // For instance, when completeing a form (as a call back) or loading the page for first time, or resetting ...
+    this.workflow = wf;
+    this.currentTask = this.workflow.next_task;
+    this.loading = false;
+    this.logTaskData(this.currentTask);
+    scrollToTop(this.deviceDetector);
+    this.location.go("workflow/" + this.workflowId + "/task/" + this.currentTask.id);
```

Comment:
again a small complaint but i think location.go() might interfere with the browser back button usability. Maybe that's less relevant when you consider when this method is used though